### PR TITLE
feat: app boilerplate

### PR DESCRIPTION
**Pull request by mhuynh5757**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

This PR replaces the former example app with a brand-new templated boilerplate using the React Native CLI.

### CODE REVIEW

**Comment by charliesantos** [comment]
@mhuynh5757 there's already some files in the [task/monorepo-restructure](https://github.com/twilio/twilio-voice-react-native-app/tree/task/monorepo-restructure) branch, the one you're submitting this PR to. I think we need to review those changes first before submitting more PRs against it. Can you please submit a PR to main branch instead so we know the source of truth?

File: app/.node-version [hunk 1]
```diff
@@ -0,0 +1 @@
+16
```
Comment (charliesantos) [inline app/.node-version:1]: Can we use the latest LTS version? I think we're at 18 now.

File: app/.prettierrc.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  arrowParens: 'avoid',
+  bracketSameLine: true,
+  bracketSpacing: false,
+  singleQuote: true,
+  trailingComma: 'all',
```
Comment (charliesantos) [inline app/.prettierrc.js:6]: Is this the default, or is this what you're proposing? I prefer `bracketSpacing` to be true...

File: app/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  extends: '@react-native-community',
+  parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint'],
+  overrides: [
+    {
+      files: ['*.ts', '*.tsx'],
+      rules: {
+        '@typescript-eslint/no-shadow': ['error'],
+        'no-shadow': 'off',
+        'no-undef': 'off',
```
Comment (charliesantos) [inline app/.eslintrc.js:12]: Is this just default or what you're proposing? I think we need to add more rules. Probably not in this PR but wanted to know your intention.

**Review by charliesantos — COMMENTED** [review]
Left a few comments. We need some eyes on the android and ios related parts. @kbagchiGWC @bobiechen-twilio .

Also, please update all references of "TwilioVoiceReactNativeReferenceApp" to "TwilioVoiceReactNativeApp". That is, no need to mention the "reference" part.

File: app/.node-version (hunk 1 above)
Comment (mhuynh5757) [reply app/.node-version:1]: This is recommended by the React Native team, but I can try with 18 from here on out. If nothing goes wrong I'll change this in future PRs.

File: app/.prettierrc.js (hunk 2 above)
Comment (mhuynh5757) [reply app/.prettierrc.js:6]: I'll change this to `true`, it was `false` by default from the RN boilerplate.

File: app/.eslintrc.js (hunk 3 above)
Comment (mhuynh5757) [reply app/.eslintrc.js:12]: This file is entirely defaults, let's change things up as we see fit. 👍

File: app/.watchmanconfig [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (mhuynh5757) [reply app/.watchmanconfig:1]: Default file included with the RN boilerplate, not sure what happens if we remove it.

**Pull request closed without merging**